content/guides/event-streaming/spring-cloud-stream-kafka-p2.md

```diff
@@ -0,0 +1,301 @@
+
+Application Configuration part ([application.yml](https://github.com/ehsaniara/scs-kafka-intro/blob/main/scs-100/src/main/resources/application.yml#L8))
+```yaml
+spring.cloud.stream.bindings.inventoryChecking-out.destination: scs-100.inventoryChecking
+```
+
+
+Our expectation from this service is to create the order and produce it in the topic. You may also separate this component into different applications and scale it. Eventually, this service can also be the only point of interaction between the front-end and back-end applications. You may refer to this component as your **Event Source**.
+
+## Inventory Service
+
+Inventory Service is also referred to as Event Processor. Here you have a consumer to your inventory check topic, and every message in this topic represents an order that needs to be checked for inventory. This is called an "Inventory Check" but it can vary in the real scenario, such as payments, supplier delivery, custom delivery, state tax calculator, etc.
+
+At this point, you can have multiples of these services for other purposes. Theoretically speaking, it doesn't matter how much time this process takes since you are not blocking the original request thread.
+
+[OrderService.java](https://github.com/ehsaniara/scs-kafka-intro/blob/main/scs-100/src/main/java/com/ehsaniara/scs_kafka_intro/scs100/OrderService.java#L68)
+
+```java
+@StreamListener(OrderBinder.INVENTORY_CHECKING_IN)
+@SneakyThrows
+public void checkInventory(@Payload Order orderIn) {
+   log.debug("checkInventory orderIn: {}", orderIn);
+   orderIn.setOrderStatus(OrderStatus.INVENTORY_CHECKING);
+   orderDataBase.put(orderIn.getOrderUuid(), orderIn);
+
+   Thread.sleep(5_000);//5 sec delay
+
+   // just a simulation of create exception for random orders (1 in 2) in case of inventory insufficiency
+   if (System.currentTimeMillis() % 2 == 0) {
+       orderIn.setOrderStatus(OrderStatus.OUT_OF_STOCK);
+       orderDataBase.put(orderIn.getOrderUuid(), orderIn);
+       log.warn("Let's assume we ran out of stock for item: {}", orderIn.getItemName());
+       Thread.sleep(5_000);//5 sec delay
+       throw new OrderFailedException(String.format("insufficient inventory for order: %s", orderIn.getOrderUuid()));
+   }
+
+   //Order is good to go for shipping
+   orderBinder.shippingOut()//
+           .send(MessageBuilder.withPayload(orderIn)//
+                   .setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.APPLICATION_JSON)//
+                   .build());
+}
+```
+_This method is the consumer and produce order event_
+
+In this method, you’ve intentionally thrown a runtime exception to simulate one of the cool features of [`SCS`](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka) which is provided out of the box, “Retry Mechanism”. It can easily be configured in application.yml with no extra coding. It also supports the “Back off and retry mechanism”. The retry value is 3 seconds, by default, and the `backoff` value is 5 seconds, by default.
```

Comment:
```suggestion
In this method, you’ve intentionally thrown a runtime exception to simulate one of the cool features of [`SCS`](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka) which is provided out of the box, “Retry Mechanism”. It can easily be configured in `application.yml` with no extra coding. It also supports the “Back off and retry mechanism”. The retry value is 3 seconds, by default, and the `backoff` value is 5 seconds, by default.
```